pkg/async/events/implementations/node_execution_event_writer.go

```diff
@@ -0,0 +1,41 @@
+package implementations
+
+import (
+	"context"
+
+	"github.com/flyteorg/flyteadmin/pkg/async/events/interfaces"
+	"github.com/flyteorg/flyteadmin/pkg/repositories"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/transformers"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+)
+
+type nodeExecutionEventWriter struct {
+	db     repositories.RepositoryInterface
+	events chan admin.NodeExecutionEventRequest
+}
+
+func (w *nodeExecutionEventWriter) Write(event admin.NodeExecutionEventRequest) {
+	w.events <- event
+}
+
+func (w *nodeExecutionEventWriter) Run() {
+	for event := range w.events {
+		eventModel, err := transformers.CreateNodeExecutionEventModel(event)
+		if err != nil {
+			logger.Warnf(context.TODO(), "Failed to transform event [%+v] to database model with err [%+v]", event, err)
+			continue
+		}
+		err = w.db.NodeExecutionEventRepo().Create(context.TODO(), *eventModel)
+		if err != nil {
+			logger.Warnf(context.TODO(), "Failed to write event [%+v] to database with err [%+v]", event, err)
+		}
+	}
+}
+
+func NewNodeExecutionEventWriter(db repositories.RepositoryInterface) interfaces.NodeExecutionEventWriter {
+	return &nodeExecutionEventWriter{
+		db:     db,
+		events: make(chan admin.NodeExecutionEventRequest),
```

Comment:
you should make a buffered channel?

Reply:
You should and the length of the channel should be pretty large, else you will block sender. Right now this is a sync send

Reply:
yes good catch, ty! not sure what a good buffer size would be, any suggestions @kumare3 ?

Reply:
done & added a test for this case